src/engine/plugin.ts

```diff
@@ -147,57 +137,33 @@ export class Plugin<T extends Api> implements PluginApi<T> {
   }
 
   /**
-   * Create an iframe element
-   * @param profile The profile of the plugin
+   * Create and return the iframe
    */
-  private async create(profile: PluginProfile) {
-    // Create
-    try {
-      this.iframe = document.createElement('iframe')
-      this.iframe.setAttribute('sandbox', 'allow-scripts')
-      this.iframe.setAttribute('seamless', 'true')
-      this.iframe.src = profile.url
-      if (profile.location) {
-        const { name, key } = profile.location
-        const message = {
-          action: 'request',
-          name,
-          key,
-          payload: { ...profile, element: this.iframe },
-        }
-        await this.request(message)
-      } else if (this.pluginLocation) {
-        this.pluginLocation.resolveLocaton(this.iframe)
-      } else {
-        document.body.appendChild(this.iframe)
-      }
-      // Wait for the iframe to load and handshake
-      this.iframe.onload = () => {
-        if (!this.iframe.contentWindow) {
-          throw new Error('No window attached to Iframe yet')
-        }
-        this.origin = new URL(this.iframe.src).origin
-        this.source = this.iframe.contentWindow
-        this.postMessage({
-          action: 'request',
-          name: this.name,
-          key: 'handshake',
-        })
-      }
-    } catch (err) {
-      console.log(err)
+  public render() {
+    if (this.iframe) {

```

Comment:
that might be a problem if one activate / deactivate / activate a plugin, cause the instance seems not reset upon deactivation. Perhaps should we keep a `deactivation` hook for reset this object

Reply:
ah i see `deactivate`, should the iframe ref should be removed to

Reply:
Yes, indeed. Thanks